spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -401,7 +418,7 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       Double.NegativeInfinity,
       0.0d) ++
       Range(0, dataSize).map(_ => r.nextDouble())
-    withNulls(values).toDF("a")
+    castTest(withNulls(values).toDF("a"), DataTypes.StringType)
```

Comment:
Fixing a regression that was introduced in https://github.com/apache/datafusion-comet/pull/346